### PR TITLE
[geoawareness][mock_uss] Implement /check endpoint

### PR DESCRIPTION
**Pull request by barroco**
This PR implements the /check endpoint of the geoawareness mock service.

### CODE REVIEW

File: monitoring/mock_uss/geoawareness/check.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+        start = a.get("startDateTime", StringBasedDateTime(datetime.min))
+        end = a.get("endDateTime", StringBasedDateTime(datetime.max))
+
+        in_range = _is_in_date_range(start, end, after, before)
+
+        schedule = a.get("schedule", None)
+        if schedule is None:
+            logger.debug(f"     * Date in range => {in_range}")
+            return in_range
```
Comment (BenjaminPelletier) [inline monitoring/mock_uss/geoawareness/check.py:159]: It seems like we only want to return if there's a match (`True`) -- if we return `False` inside the `feature.applicability` loop, it seems we'd be skipping some of the possible applicability entries.

File: monitoring/monitorlib/geoawareness_automated_testing/api.py [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
+    VerticalReferenceType,
+    Restriction,
+)
 
 SCOPE_GEOAWARENESS_TEST = "geo-awareness.test"
 
 # Mirrors of types defined in geo-awareness automated testing API
 
 
+class Position(ImplicitDict):
```
Comment (BenjaminPelletier) [inline monitoring/monitorlib/geoawareness_automated_testing/api.py:15]: We don't have to migrate all the existing data types right away, but could we import any of the new ones from `uas_standards.interuss.automated_testing.geo_awareness.v1.api`?

File: monitoring/mock_uss/geoawareness/check.py (hunk 1 above)
Comment (barroco) [reply monitoring/mock_uss/geoawareness/check.py]: Thanks for spotting it. fixed.

File: monitoring/mock_uss/geoawareness/check.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+
+    for a in feature.applicability:
+        if a.permanent == YESNO.YES:
+            logger.debug(f"     * Permanent applicability => True")
+            return True
+
+        start = a.get("startDateTime", StringBasedDateTime(datetime.min))
+        end = a.get("endDateTime", StringBasedDateTime(datetime.max))
+
+        in_range = _is_in_date_range(start, end, after, before)
```
[the remark this answers is not in the document]
Comment (barroco) [reply monitoring/mock_uss/geoawareness/check.py]: Much more elegant with the continue, thanks.

File: monitoring/mock_uss/geoawareness/check.py [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+                logger.debug(
+                    f" {j+1}. Source {source_id} is not ready ({source.state}). Skip."
+                )
+                continue
+
+            fmt = source.definition.https_source.format
+            if fmt == HttpsSourceFormat.Ed269:
+                logger.debug(f" {j+1}. ED269 source {source_id} ready.")
+                result = combine_results(
+                    result, evaluate_source(source, check.filterSets)
```
[the remark this answers is not in the document]
Comment (barroco) [reply monitoring/mock_uss/geoawareness/check.py:50]: Format dependent functions have been moved to a dedicated package.

File: monitoring/mock_uss/geoawareness/check.py [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+
+    # Evaluate timing
+    timing_match = evaluate_timing(
+        feature, filter_set.get("after", None), filter_set.get("before", None)
+    )
+    if not timing_match:
+        logger.debug(f"    {feature.identifier}: Timing not matched - Absent")
+        return False
+
+    # Evaluate U-Space
```
[the remark this answers is not in the document]
Comment (barroco) [reply monitoring/mock_uss/geoawareness/check.py]: Done, much clearer, thanks

File: monitoring/monitorlib/geoawareness_automated_testing/api.py (hunk 2 above)
Comment (barroco) [reply monitoring/monitorlib/geoawareness_automated_testing/api.py]: All data types should have been migrated by now.

**Review by BenjaminPelletier — APPROVED** [review]

**Pull request merged**